raiden-ts/src/helpers.ts

```diff
@@ -72,24 +104,33 @@ export const getSigner = async (account: string | number | Signer, provider: Jso
     } else {
       throw new Error(`Signer ${account} not connected to ${provider}`);
     }
+    address = (await signer.getAddress()) as Address;
   } else if (typeof account === 'number') {
     // index of account in provider
     signer = provider.getSigner(account);
+    address = (await signer.getAddress()) as Address;
   } else if (Address.is(account)) {
     // address
     const accounts = await provider.listAccounts();
     if (!accounts.includes(account)) {
       throw new Error(`Account "${account}" not found in provider, got=${accounts}`);
     }
     signer = provider.getSigner(account);
+    address = account;
   } else if (PrivateKey.is(account)) {
     // private key
     signer = new Wallet(account, provider);
+    address = signer.address as Address;
   } else {
     throw new Error('String account must be either a 0x-encoded address or private key');
   }
 
-  return signer;
+  if (subkey) {
```

Comment:
Can't we skip all the stuff above if `subkey === true`?

Reply:
No, `main` key still needs to be generated/instantiated as usual. The `subkey` is for after we have a main `Signer` already, however we got it, and then derive it, put the main on `main`, and replace the _actual_ with the subkey's.